### ISSUE TITLE
MP4 results in MPEG-TS instead of MPEG-4

### ISSUE THREAD

**Issue by Anuskuss** [yt-dlp/yt-dlp#1701]
### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2021.11.10.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Description

Downloading HLS videos (without audio) and setting the extension of the output to `.mp4` should give you a valid MP4 (MPEG-4), but instead it gives you a MPEG-TS. Trying to force it via `--merge-output-format mp4` or `--recode-video mp4` doesn't work:
```
[ffmpeg] Not converting video file video.mp4 - already is in target format mp4
```
Most players can handle MPEG-4 AVC video in a MPEG-TS container, but web browsers don't, so my downloaded files won't play as HTML5 videos (`<video src="video.mp4></video>`). The workaround is either downloading DASH, or running them through `ffmpeg` afterwards. Ideally, yt-dlp would just create compliant MP4s.

### Verbose log
Commands were done with `youtube-dl` but I've verified that the behaviour hasn't changed: https://github.com/ytdl-org/youtube-dl/issues/26410#issuecomment-970672946

**Comment by nao20010128nao** [comment]
That's right native downloader for HLS just concatenate all segments and usually no conversion for real MP4 file.

To devs: the name looks odd, but `FFmpegFixupDurationPP` fixes this

**Comment by Anuskuss** [comment]
Confirmed working. Thanks @pukkandan

**Comment by pukkandan** [comment]
> Seems like I can't quite get rid of my custom ffmpeg command because I still need -movflags +faststart. Apparently it has already been implemented in 06167fb, but it doesn't do it for me. 

Can you explain?

**Comment by Anuskuss** [comment]
I'm assuming that you already know what `faststart` does, right? Since it's not really related to this issue, you should check out #1491 where I left two commands which should've given me "streamable" MP4s, but they don't for some reason.
